util/argo/argo.go

```diff
@@ -552,6 +552,21 @@ func ContainsSyncResource(name string, namespace string, gvk schema.GroupVersion
 	return false
 }
 
+// IncludeResource checks if an app resource matches atleast one of the filters, then it returns true.
+func IncludeResource(resourceName string, resourceNamespace string, gvk schema.GroupVersionKind,
```

Comment:
In other places where we use `!<filter>` the rule is that at least one positive filter must match and _no_ negative filter must match. Is that the behavior here?

Reply:
Yes. The various filter combinations are mentioned in the comments of each test scenario in the unit test case. You and check and let me know if my understanding is right.

Reply:
I don't think any of the existing unit tests cover this case.

The test would be something like:

1) include all Service resources
2) exclude this specific Service resource (say, `service1`)

We'd expect from a set of `service1`, `service2`, `deployment1` that the only returned resource would be `service1`.

Reply:
I think that, in the case where _only_ exclude filters are provided, we can add an implicit "include everything" filter.

But if one or more "include" filters is provided, we should not use an implicit include-all filter.

Reply:
Both cases are covered as part of unit test cases, 1 - [here](https://github.com/argoproj/argo-cd/pull/10893/files#diff-09dc2060ef1815120b6ffc55153594e558691f94bd2fcd47e0e0acb8ba68066aR1079) and 2 - [here](https://github.com/argoproj/argo-cd/pull/10893/files#diff-09dc2060ef1815120b6ffc55153594e558691f94bd2fcd47e0e0acb8ba68066aR1064)

Reply:
Yes, when only exclude filters are provided (Ex: [here](https://github.com/argoproj/argo-cd/pull/10893/files#diff-09dc2060ef1815120b6ffc55153594e558691f94bd2fcd47e0e0acb8ba68066aR1059)), it acts like an OR condition and everything is included. And when only 'include' filters are provided, only those resources are listed (Ex: [here](https://github.com/argoproj/argo-cd/pull/10893/files#diff-09dc2060ef1815120b6ffc55153594e558691f94bd2fcd47e0e0acb8ba68066aR1049))